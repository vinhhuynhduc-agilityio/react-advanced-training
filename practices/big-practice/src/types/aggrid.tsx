import { ColDef } from "ag-grid-community";
import { RowData } from "./users";

export interface DataGridProps {
  rowData: RowData[];
  columnDefs: ColDef<RowData>[];
};