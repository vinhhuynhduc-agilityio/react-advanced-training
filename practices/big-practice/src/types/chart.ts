
export interface FormattedDataTotalTasksCompleted {
  month: string;
  [year: number]: number;
};

export interface FormattedDataTotalTasksByProjects {
  projectName: string;
  [year: number]: number;
};
