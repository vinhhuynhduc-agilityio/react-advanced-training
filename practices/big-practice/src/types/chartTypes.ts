export interface FormattedMonthData {
  month: string;
  2023: number;
  2024: number;
};

export interface FormattedProjectData {
  projectName: string;
  2023: number;
  2024: number;
}
