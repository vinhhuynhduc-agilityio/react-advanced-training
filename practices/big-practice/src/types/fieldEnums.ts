export  enum FieldType {
  PROJECT = 'project',
  USER = 'user',
  TASK_NAME = 'taskName',
};
