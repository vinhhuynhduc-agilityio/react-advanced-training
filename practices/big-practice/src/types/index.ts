export * from './chart';
export * from './task';
export * from './user';
export * from './project';
