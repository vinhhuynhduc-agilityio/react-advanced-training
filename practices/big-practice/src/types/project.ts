export interface ProjectsData {
  id: string;
  projectName: string;
};
