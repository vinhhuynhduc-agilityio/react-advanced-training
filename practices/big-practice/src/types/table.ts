export interface UserData {
  id: string;
  fullName: string;
  earnings: string;
  email: string;
  avatarUrl: string;
  registered: string;
  lastUpdated: string;
}

export interface TaskData {
  id: string;
  userId: string;
  projectId: string;
  taskName: string;
  startDate: string;
  completedDate: string;
  currency: number;
  status: boolean;
  projectName: string;
  fullName: string;
  tooltipShowDelay?: number;
}

export interface ProjectsData {
  id: string;
  projectName: string;
}

export interface UserListDrawerProps {
  users: UserData[];
  selectedUserId: string | null;
  onUserSelected: (userId: string | null) => void;
  sourceComponent: string | null;
}

export type StatusWithDate = {
  status: boolean;
  completedDate: string;
};

export type FieldValue =
  | string
  | number
  | boolean
  | ProjectsData
  | UserData
  | StatusWithDate;
