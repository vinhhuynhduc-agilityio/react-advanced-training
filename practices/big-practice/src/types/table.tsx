import {
  ColDef,
  GetRowIdParams,
  GridReadyEvent,
  RowClassParams,
  RowClickedEvent
} from "ag-grid-community";

export interface UserData {
  id: string;
  fullName: string;
  earnings: string;
  email: string;
  avatarUrl: string;
  registered: string;
  lastUpdated: string;
};

export interface TaskData {
  id: string;
  userId: string;
  projectId: string;
  taskName: string;
  startDate: string;
  completedDate: string;
  currency: number;
  status: boolean;
  projectName: string;
  fullName: string;
};

export interface ProjectsData {
  id: string;
  projectName: string;
};

export interface DataGridProps<T> {
  rowData: T[];
  userData?: T[];
  columnDefs: ColDef<T>[];
  onRowClicked?: (event: RowClickedEvent) => void;
  getRowClass?: (params: RowClassParams) => string;
  rowHeight?: number;
  onGridReady: (event: GridReadyEvent) => void;
  getRowId: (params: GetRowIdParams<T>) => string;
};

export interface UserListDrawerProps {
  users: UserData[];
  selectedUserId: string | null;
  onUserSelected: (userId: string | null) => void;
  sourceComponent: string | null;
};

export interface TaskDataProps {
  tasks: TaskData[];
  selectedUserId: string | null;
  onTaskRowSelected: (userId: string | null) => void;
  projects: ProjectsData[];
  sourceComponent: string | null;
  users: UserData[]
};
