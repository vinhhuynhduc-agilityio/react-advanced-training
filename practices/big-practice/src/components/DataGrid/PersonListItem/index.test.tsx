import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ICellRendererParams } from 'ag-grid-community';
import { PersonListItem } from '.';
import { UserData } from '@/types';

const mockData: UserData = {
  id: "d290f1ee-6c54-4b01-90e6-d701748f0851",
  fullName: "Joe Bloggs",
  earnings: "$5000",
  email: "john@example.com",
  avatarUrl: "https://i.pravatar.cc/150?img=1",
  registered: "May 21, 2020 17:02:06",
  lastUpdated: "October 10, 2023 12:30:00"
};

describe('PersonListItem component', () => {
  it('matches snapshot for default state', () => {
    const params: ICellRendererParams<UserData> = { data: mockData } as ICellRendererParams<UserData>;
    const { container } = render(<PersonListItem {...params} />);
    expect(container).toMatchSnapshot();
  });

  it('should render user data correctly', () => {
    const params: ICellRendererParams<UserData> = { data: mockData } as ICellRendererParams<UserData>;

    render(<PersonListItem {...params} />);

    expect(screen.getByText('Joe Bloggs')).toBeInTheDocument();
    expect(screen.getByText('$5000')).toBeInTheDocument();
    const img = screen.getByAltText('avatar') as HTMLImageElement;
    expect(img.src).toBe('https://i.pravatar.cc/150?img=1');
  });

  it('should return null when params.data is undefined', () => {
    const params: ICellRendererParams<UserData> = { data: undefined } as ICellRendererParams<UserData>;
    const { container } = render(<PersonListItem {...params} />);

    expect(container.firstChild).toBeNull();
  });
});
