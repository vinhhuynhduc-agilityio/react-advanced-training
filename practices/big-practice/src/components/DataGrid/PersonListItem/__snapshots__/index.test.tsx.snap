// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonListItem component matches snapshot for default state 1`] = `
<div>
  <div
    aria-label="Joe Bloggs"
    class="flex items-center py-2"
    role="listitem"
  >
    <img
      alt="User Avatar"
      aria-label=""
      class="rounded-full object-cover w-14 h-14"
      src="https://i.pravatar.cc/150?img=1"
    />
    <div
      class="flex flex-col ml-4"
    >
      <div
        class="text-base font-medium text-[#313131]"
      >
        Joe Bloggs
      </div>
      <div
        class="text-base font-normal text-[#475466]"
      >
        $5000
      </div>
    </div>
  </div>
</div>
`;
